oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleConfig.java

```diff
@@ -134,10 +134,31 @@
     @Getter
     private String searchableLogsTags = "";
 
+    /**
+     * The number of threads used to synchronously refresh the metrics data to the storage. When the value is zero or
+     * negative, the number of processors will used as the default value.
+     *
+     * @since 8.5.0
+     */
+    @Setter
+    private int syncThreads = -1;
+
+    /**
+     * The maximum number of processes supported for each synchronous storage operation. When the number of the flush
+     * data is greater than this value, it will be assigned to multiple cores for execution.
+     */
+    @Getter
+    @Setter
+    private int maxSyncOperationNum = 50000;
+
     public CoreModuleConfig() {
         this.downsampling = new ArrayList<>();
     }
 
+    public int getSyncThreads() {
+        return this.syncThreads > 0 ? this.syncThreads : Runtime.getRuntime().availableProcessors();
```

Comment:
Why choose the number of the CPU core? Any benchmark to support this?

Reply:
We have the straming based on this, including level 1 and level 2.
Why this is determined by this too?

Reply:
I just set the maximum parallelism for this. But I have no experience with this.

Reply:
But as default value, mostly would be the real configuration for many users. I would say this is dangerous.

This value should relate to ES server CPU, memory and scale, rather than oap. Right?

Reply:
You are right. Any suggestions? Maybe the default value is 1 or 2?

Reply:
I am thinking about 1. Let's wait for @zifeihan feedback. As he seems having some private version experimenting this.